compaction/compaction.cc

```diff
@@ -1125,7 +1125,7 @@ class cleanup_compaction final : public regular_compaction {
         : cleanup_compaction(table_s, std::move(descriptor), cdata, opts.owned_ranges) {}
 
     flat_mutation_reader make_sstable_reader() const override {
-        return make_filtering_reader(regular_compaction::make_sstable_reader(), make_partition_filter());
+        return downgrade_to_v1(make_filtering_reader(upgrade_to_v2(regular_compaction::make_sstable_reader()), make_partition_filter()));
```

Comment:
sstable reader already is a V2 reader so maybe we could use it directly here without downgrading it to v1 and upgrading again before passing to filtering reader.

Reply:
we have to adjust a few interfaces for this to work. I'll proceed with this work soon.

Reply:
I meant doing `downgrade_to_v1(make_filtering_reader(regular_compaction::make_sstable_reader_v2())`
That shouldn't require changes to any consumers.

Reply:
make_sstable_reader() calls sstable_set::make_local_shard_sstable_reader() which is still confined to v1. I am planning to convert everything in compaction and their dependencies like sstable_set, so we won't have to play these games anymore.